migrate/20241213_archived_record.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+Sequel.migration do
+  change do
+    create_table(:archived_record, partition_by: :deleted_at, partition_type: :range) do
```

Comment:
This may be a dumb question, as I haven't used deleted_records myself, but isn't the biggest issue with deleted_records that you have to do sequential scans?  What are your thoughts on creating an index, maybe on deleted_at or model_name (depending on which is more valuable)?  I realize you give up cheaper inserts, but it should make querying the table faster.

Reply:
I believe the vast majority of queries would look at a single type of model in a given time interval. I added an index on (model_name, deleted_at), which should work well if there's a filter on both model_name and deleted_at or just model_name. If theres a filter on deleted_at only, which I believe would be rare, we'd still do a sequential scan, but at least only on relevant partitions. I'm fine with changing the index if I'm misjudging query access patterns (if we not almost always filter by model_name).